src/anemoi/graphs/nodes/builders/from_file.py

```diff
@@ -52,15 +52,52 @@ def get_coordinates(self) -> torch.Tensor:
 
 
 class CutOutZarrDatasetNodes(ZarrDatasetNodes):
-    """Nodes from Zarr dataset."""
+    """Nodes from Zarr dataset.
+
+    Attributes
+    ----------
+    lam_dataset : str
+        The limited area dataset.
+    forcing_dataset : str
+        The forcing dataset.
+    thinning : int, optional
+        The thinning factor. Defaults to 1, which means no thinning.
+    forcing_area : list[float], optional
+        The area of the forcing dataset. Specify the longitude and
+        latitudes boundaries as (north, west, south, east). Defaults
+        to None, which means the forcing dataset is not cropped.
+
+    Methods
+    -------
+    get_coordinates()
+        Get the lat-lon coordinates of the nodes.
+    register_nodes(graph, name)
+        Register the nodes in the graph.
+    register_attributes(graph, name, config)
+        Register the attributes in the nodes of the graph specified.
+    update_graph(graph, name, attr_config)
+        Update the graph with new nodes and attributes.
+    """
 
     def __init__(
-        self, name: str, lam_dataset: str, forcing_dataset: str, thinning: int = 1, adjust: str = "all"
+        self,
+        name: str,
+        lam_dataset: str,
+        forcing_dataset: str,
+        thinning: int = 1,
+        forcing_area: list[float] | None = None,
+        adjust: str = "all",
     ) -> None:
-        dataset_config = {
-            "cutout": [{"dataset": lam_dataset, "thinning": thinning}, {"dataset": forcing_dataset}],
-            "adjust": adjust,
-        }
+        lam_config = {"dataset": lam_dataset, "thinning": thinning}
+        forcing_config = {"dataset": forcing_dataset}
```

Comment:
Yes, we could do it. My reasoning behind this approach was to make it easier to use for a user who does not know what anemoi-dataset needs as input.

Reply:
Personally I would prefer if the config yaml that anemoi-graphs takes as input as of exactly the same style as anemoi-training takes (under /graph) and furthermore that if one specifies a dataset to build the nodes through CutOutZarrDatasetNodes that the dataset specification in the config yaml is identical to the config/dictionary style one uses in anemoi-datasets. Right now this is not really the case. Maybe a user will need to learn a bit more formatting, but this will help him later to use anemoi-datasets and vice versa. Currently a user that uses both needs to remember various types of formatting to get the same result, depending if he uses anemoi-datasets, anemoi-training or anemoi-graphs.